silq/instrument_interfaces/lecroy/ArbStudio1104_interface.py

```diff
@@ -420,20 +420,21 @@ def implement(self, sampling_rates, input_pulse_sequence, plot=False, **kwargs):
                 output channel, where each element in sequence indicates the
                 waveform that must be played after the trigger
         """
-        # Find all trigger pulses occuring within this pulse
+        # Find all trigger pulses occurring within this pulse
         trigger_pulses = input_pulse_sequence.get_pulses(
             t_start=('>', self.pulse.t_start),
             t_stop=('<', self.pulse.t_stop),
             trigger=True)
         assert len(trigger_pulses) == 0, \
-            "Cannot implement multi sine pulse if the arbstudio receives intermediary triggers"
+            "Cannot implement multi sine pulse if the Arbstudio receives intermediary triggers"
 
         if isinstance(self.pulse.connection, SingleConnection):
             channels = [self.pulse.connection.output['channel'].name]
         else:
             raise Exception(f"No implementation for connection {self.pulse.connection}")
 
-        assert [frequency > 0 for frequency in self.pulse.frequencies], "All pulse frequencies must be larger than zero."
+        assert [frequency > 0 for frequency in self.pulse.frequencies], \
+            "All pulse frequencies must be larger than zero."
```

Comment:
If the ArbStudio is providing IQ modulation, then this prevents you from using the lower sideband and the carrier (frequency <= 0).
I don't think this is what you want.

Reply:
well, generally we need just one sideband, which if IQ modulation is properly calibrated would be the only one giving power... unless I am missing something... Plus if we allow negative frequencies, I am not sure how it will work... sin(-|f|*t) - sounds confusing to me... I never worked with IQ modulation before, so don't have much experience... if something is off I am happy to listen if you have time to explain (WhatsApp/Teams or whatever is good for you)

Reply:
I'll try to answer this here, but if it's still unclear then we can have a chat.

Basically I'm saying that you can use IQ modulation to target either the upper or lower sideband, offset from the carrier frequency by the frequency of the IQ terms, depending on whether the IQ terms are rotating clockwise or counter-clockwise. 

So the microwave interface tries to implement a `MultiSinePulse` it will create additional pulses for the ArbStudio, which are also `MultiSinePulse`s. In the `MultiSinePulseImplementation` of the ArbStudio, you assert that the frequencies of each component Sine must be greater than 0, but this prevents you from utilising the clockwise-type of IQ modulation, i.e. it prevents you from modulating the microwave output frequency below that of the carrier frequency. You've also prevented having a constant term (i.e. frequency = 0) which would output at the carrier frequency.

I understand that you generally only use the upper sideband, but I see no reason to impose this limitation in the driver. If you only want to use the upper sideband, then you can just set your carrier frequency accordingly.

Reply:
Just to confirm I understand correctly - if the frequency w < 0, then our the main modulated signal will be just on the lower sideband instead of the upper sideband, i.e. we just switch from sin(w_LO+w) to sin(w_LO-|w|)? If so, then I agree, ok, thanks, will correct this.

Reply:
Corrected.